storage/bucket.go

```diff
@@ -59,7 +59,8 @@ func (b *FileReaderAt) ReadAt(p []byte, off int64) (n int, err error) {
 	for total < len(p) { // Read does not guarantee the buffer will be full, but ReadAt does
 		n, err = rc.Read(p[total:])
 		total += n
-		if err != nil {
+		// If io.EOF is returned it means we read the end of the file and simply return the total
+		if err != nil && err != io.EOF {
```

Comment:
nit: I prefer using `errors.Is` (it's more complete). I also have a minor suggestion to make this code clearer
```suggestion
		if err != nil {
		    if errors.Is(err, io.EOF) {
		        // If io.EOF is returned it means we read the end of the file and simply return the total.
		        break
		    }
```

Reply:
I was wondering if we should use the newer APIs. Thanks!